### PR TITLE
Add data-testids to side nav, action_menu and dialogs

### PR DESCRIPTION
**Pull request by cornshuckin2**
This adds generic testids for the automation team to use for side navigation, action menu and dialogs.

### CODE REVIEW

File: app/components/anchor/dialog_component.rb [hunk 1]
```diff
@@ -2,7 +2,8 @@ module Anchor
   class DialogComponent < Component
     renders_one :show_button, ->(**kwargs) do
       ButtonComponent.new(
-        data: { action: "dialog#showModal" },
+        data: { action: "dialog#showModal",
+                testid: "#{title.parameterize.gsub('_', '-')}-btn" },
```
Comment (cornshuckin2) [inline app/components/anchor/dialog_component.rb:6]: `dasherize` would not work here, not really sure why if rails folks would have more insight

Comment (dgalarza) [reply app/components/anchor/dialog_component.rb:6]: hmm @cornshuckin2 do you know what kind of error you were getting?

Comment (cornshuckin2) [reply app/components/anchor/dialog_component.rb:6]: No error, the resulting testid would just not be formatted

**Review by dgalarza — APPROVED** [review]

**Review by tysongach — COMMENTED** [review]
Seems fine. I’m not the biggest fan of these test IDs being here as they have direct consequences to tests in another project completely.

It would make me feel a little better if we could be more systematic with the approach. For example, if they were always generated (not written by hand) and we were consistent about it throughout the whole library (every component and every slot gets a `testid`).

**Review by clawhun — APPROVED** [review]

**Review by Simonpedro — APPROVED** [review]

File: app/components/anchor/dialog_component.rb [hunk 2]
```diff
@@ -2,7 +2,8 @@ module Anchor
   class DialogComponent < Component
     renders_one :show_button, ->(**kwargs) do
       ButtonComponent.new(
-        data: { action: "dialog#showModal" },
+        data: { action: "dialog#showModal",
+                testid: "#{title.parameterize.dasherize}-btn" },
```
[the remark this answers is not in the document]
Comment (cornshuckin2) [reply app/components/anchor/dialog_component.rb]: Fixed, TY!

**Review by tysongach — APPROVED** [review]
Spoke with @cornshuckin2 - going to run with this for now and I’m going to investigate automatic generation of `testid`s for every component and their slots, if any.

**Comment by tysongach** [comment]
@cornshuckin2 I can cut a release after this is merged.

**Comment by cornshuckin2** [comment]
@tysongach Merged, appreciate it

**Comment by tysongach** [comment]
@cornshuckin2 v0.7.0 is now available!

**Pull request merged**